Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -0,0 +1,361 @@
+
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(logger, localNuGetSourceDir, localNuGetDownloadDir);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            bool result = apg.Generate(new Roslyn.CommandLine.NuGetReference(Node.A.ToString(), new SemanticVersion("1.0")), "cs", null, outputDir);
+
+            UninstallAllTestNuGetPackages(localNuGetSourceStore);
+
+            // Assert
+            Assert.IsFalse(result, "Generator should fail if a package dependency requires license accept");
+        }
+
+        #region Private methods
+
+        private void UninstallAllTestNuGetPackages(IPackageManager manager)
+        {
+            foreach (Node node in Enum.GetValues(typeof(Node)))
+            {
+                manager.UninstallPackage(node.ToString(), new SemanticVersion("1.0"), true, false);
+            }
+        }
+
+        /// <summary>
+        /// Creates a graph used for testing, with nodes labelled in breadth-first order.
+        /// 
+        /// Visually:
+        /// A
+        /// |\
+        /// B C 
+        /// | |\
+        /// D E F
+        /// </summary>
+        /// <param name="nodesRequireLicense">
+        /// Nodes in the graph that should be packages with the field requireLicenseAccept set to true
+        /// </param>
+        private void SetupTestGraph(IPackageManager mgr, params Node[] nodesRequireLicense)
+        {
+            // leaf nodes
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.D) ? License.Required : License.NotRequired, Node.D.ToString());
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.E) ? License.Required : License.NotRequired, Node.E.ToString());
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.F) ? License.Required : License.NotRequired, Node.F.ToString());
+
+            // non-leaf nodes
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.B) ? License.Required : License.NotRequired, Node.B.ToString(),
+                Node.D.ToString());
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.C) ? License.Required : License.NotRequired, Node.C.ToString(),
+                Node.E.ToString(), Node.F.ToString());
+
+            // root
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.A) ? License.Required : License.NotRequired, Node.A.ToString(),
+                Node.B.ToString(), Node.C.ToString());
+        }
+
+        private IPackageManager CreatePackageManager(string packageSource)
+        {
+            IPackageRepository repo = PackageRepositoryFactory.Default.CreateRepository(packageSource);
+            PackageManager mgr = new PackageManager(repo, packageSource);
+
+            return mgr;
+        }
+
+        private void CreatePackage(IPackageManager manager, License requiresLicenseAccept, string id, params string[] dependencyIds)
```

Comment:
The calling code would be cleaned if you passed the dependency ids typed as enums.